src/Storage/AssetMapCache.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace Torr\Assets\Storage;
+
+use Symfony\Bundle\FrameworkBundle\Console\Application;
+use Symfony\Component\Console\Input\ArrayInput;
+use Psr\Cache\CacheItemPoolInterface;
+use Symfony\Component\Console\Output\NullOutput;
+use Symfony\Component\HttpKernel\KernelInterface;
+use Torr\Assets\Command\AssetsDumpCommand;
+
+final class AssetMapCache
+{
+	private const CACHE_KEY = '21torr.asset.asset_map';
+
+	private CacheItemPoolInterface $cachePool;
+
+	private KernelInterface $kernel;
+
+	private AssetStorage $storage;
+
+	/**
+	 * AssetMapCache constructor.
+	 */
+	public function __construct(
+		CacheItemPoolInterface $cachePool,
+		KernelInterface $kernel,
+		AssetStorage $storage
+	)
+	{
+		$this->cachePool = $cachePool;
+		$this->kernel = $kernel;
+		$this->storage = $storage;
+	}
+
+	/**
+	 * save AssetMap in Cache
+	 */
+	public function setMapCache(AssetMap $map) : void
+	{
+		$cacheItem = $this->cachePool->getItem(self::CACHE_KEY);
+		$cacheItem->set($map);
+		$this->cachePool->save($cacheItem);
+	}
+
+	/**
+	 * get AssetMap from Cache
+	 */
+	public function getMapCache() : ?AssetMap
+	{
+		$cacheItem = $this->cachePool->getItem(self::CACHE_KEY);
+
+		if($cacheItem->isHit()) {
+			return $cacheItem->get();
+		}
+
+		return null;
+	}
+
+	/**
+	 * get AssetMap
+	 * in debug mode return new empty AssetMap and clear dump directory
+	 */
+	public function getAssetMap() : ?AssetMap
+	{
+		if($this->kernel->isDebug() === true) {
+			$this->storage->clearStorage();
+			return new AssetMap();
+		}
+
+		$assetMap = $this->getMapCache();
+
+		// If no cache exist, execute the DumpCommand
+		if(null === $assetMap) {
+			$this->executeDumpCommand();
+			$assetMap = $this->getMapCache();
+		}
+
+		return $assetMap;
+	}
+
+	/**
+	 * clear AssetMap Cache
+	 */
+	public function clearCache() : void
+	{
+		$this->cachePool->deleteItem(self::CACHE_KEY);
+	}
+
+	/**
+	 * execute AssetsDumpCommand
+	 */
+	private function executeDumpCommand() : void
```

Comment:
You should basically never call CLI commands in your application. Just fetch the service the CLI command uses internally and call it directly

Reply:
Changes are made in a new branch